File: admin/code/code.go
package code

type Language int

const (
	ZH Language = iota
	EN
)

type ErrCode int

var (
	language      = ZH
	debug         = true
	defaultErrMsg = "失败"
	msg           = map[Language]map[ErrCode]string{
		ZH: zhMsg,
		EN: enMsg,
	}
)

const (
	Success ErrCode = 0
	Error   ErrCode = 1

	ReadContextRequestBodyFailed ErrCode = 100001

	AuthTokenFailed         ErrCode = 200001 // 鉴权
	AuthTokenInvalid        ErrCode = 200002 // BearerToken无效
	AuthTokenInspectInvalid ErrCode = 200003 // BearerToken无效
	AuthTokenInfoInvalid    ErrCode = 200004 // Token信息无效

	RequestBodyInvalid   ErrCode = 300001 // 请求参数无效
	RequestQueryInvalid  ErrCode = 300002 // 请求参数无效
	RequestParamsInvalid ErrCode = 300003 // 请求参数无效

	AdminAccountPasswordInvalid ErrCode = 400001 // 密码错误
)

func SetLanguage(c Language) {
	language = c
}

func SetDebug(d bool) {
	debug = d
}

func SetDefaultErrMsg(msg string) {
	defaultErrMsg = msg
}

func GetMsgByCode(code ErrCode) string {
	if val, ok := msg[language]; ok {
		if v, ok := val[code]; ok {
			return v
		}
	}
	return defaultErrMsg
}

File: admin-vue3x/src/api/code.ts
/** 请求成功状态码 */
export const SUCCESS = 0;

/** 鉴权相关错误码（与后端 code_proto.ErrorCode 保持一致） */
export const AuthErrorCode = {
  /** 鉴权失败（未登录或令牌已过期） */
  AuthTokenFailed: 200001,
  /** BearerToken 无效 */
  AuthTokenInvalid: 200002,
  /** 登录令牌检查失败 */
  AuthTokenInspectInvalid: 200003,
  /** Token 信息无效 */
  AuthTokenInfoInvalid: 200004,
  /** 没有访问权限 */
  AuthTokenForbidden: 200005,
} as const;

/** 是否令牌失效类错误（需要重新登录） */
export const IsAuthTokenInvalidCode = (code?: number) =>
  code === AuthErrorCode.AuthTokenFailed ||
  code === AuthErrorCode.AuthTokenInvalid ||
  code === AuthErrorCode.AuthTokenInspectInvalid ||
  code === AuthErrorCode.AuthTokenInfoInvalid;

/** 是否“没有访问权限”（仅提示，不退出登录） */
export const IsAuthForbiddenCode = (code?: number) => code === AuthErrorCode.AuthTokenForbidden;

/** 业务错误状态码 */
export const ErrorCode = {
  SUCCESS,
};
